store/tx_manager.go

```diff
@@ -17,44 +20,51 @@ const defaultGasLimit uint64 = 500000
 // the local Config for the application, and the database.
 type TxManager struct {
 	*EthClient
-	KeyStore *KeyStore
-	Config   Config
-	ORM      *models.ORM
+	keyStore      *KeyStore
+	config        Config
+	orm           *models.ORM
+	activeAccount *ActiveAccount
 }
 
 // CreateTx signs and sends a transaction to the Ethereum blockchain.
 func (txm *TxManager) CreateTx(to common.Address, data []byte) (*models.Tx, error) {
-	account, err := txm.KeyStore.GetAccount()
-	if err != nil {
-		return nil, err
-	}
-	nonce, err := txm.GetNonce(account.Address)
-	if err != nil {
-		return nil, err
-	}
-	tx, err := txm.ORM.CreateTx(
-		account.Address,
-		nonce,
-		to,
-		data,
-		big.NewInt(0),
-		defaultGasLimit,
-	)
-	if err != nil {
-		return nil, err
+	if txm.activeAccount == nil {
+		return nil, errors.New("Must activate an account before creating a transaction")
 	}
+
 	blkNum, err := txm.GetBlockNumber()
 	if err != nil {
 		return nil, err
 	}
 
-	gasPrice := &txm.Config.EthGasPriceDefault
-	_, err = txm.createAttempt(tx, gasPrice, blkNum)
-	if err != nil {
-		return tx, err
-	}
+	var tx *models.Tx
+	txm.activeAccount.GetAndIncrementNonce(func(nonce uint64) error {
```

Comment:
Looks like `GetAndIncrementNonce` is returning an error we're not handling.

Reply:
On second pass, it looks like `err` is assigned to the variable that exists outside the function... cool solution, but a little unintuitive. Maybe still assign the error at the end of the function?

Reply:
The reason I went with the closure assignment approach is that I wanted the `Tx` & `error` assignment to be the same. I had initially wanted `GetAndIncrementNonce` to be generic and not be forced into returning a `Tx` so that it could be used in another context. But perhaps that's not a realistic case and I should just refactor so that it returns `Tx, error`.